packages/allure-playwright/src/index.ts

```diff
@@ -623,9 +623,11 @@ export class AllureReporter implements ReporterV2 {
     } else if (!existsSync(attachment.path!)) {
       return;
     } else {
-      // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+      const contentType =
+        attachment.name === "trace" ? "application/vnd.allure.playwright-trace" : attachment.contentType;
```

Comment:
Check the attachment content type too to be sure because we can have `trace.jpg`, `trace.js` etc.:

`attachment.name === "trace" && attachment.contentType === "application/zip" ? ...`